crates/light-client-prover/src/circuit.rs

```diff
@@ -99,35 +105,52 @@ pub fn run_circuit<DaV: DaVerifier, G: ZkvmGuest>(
             if let Ok(data) = data {
                 match data {
                     DaDataLightClient::Complete(proof) => {
-                        let journal =
-                            G::extract_raw_output(&proof).expect("DaData proofs must be valid");
-                        // TODO: select output version based on the spec
-                        let batch_proof_output: BatchProofCircuitOutput<DaV::Spec, [u8; 32]> =
-                            match G::verify_and_extract_output(
-                                &journal,
-                                &batch_proof_method_id.into(),
-                            ) {
-                                Ok(output) => output,
-                                Err(_) => continue,
-                            };
-
-                        // Do not add if last l2 height is smaller or equal to previous output
-                        // This is to defend against replay attacks, for example if somehow there is the script of batch proof 1 we do not need to go through it again
-                        if batch_proof_output.last_l2_height <= last_l2_height {
+                        if process_complete_proof::<DaV, G>(
+                            proof,
+                            batch_proof_method_id,
+                            last_l2_height,
+                            &mut initial_to_final,
+                        )
+                        .is_err()
+                        {
                             continue;
                         }
-
-                        recursive_match_state_roots(
-                            &mut initial_to_final,
-                            &BatchProofInfo::new(
-                                batch_proof_output.initial_state_root,
-                                batch_proof_output.final_state_root,
-                                batch_proof_output.last_l2_height,
-                            ),
-                        );
                     }
-                    DaDataLightClient::Aggregate(_) => todo!(),
-                    DaDataLightClient::Chunk(_) => todo!(),
+                    DaDataLightClient::Aggregate(tx_ids) => {
+                        let existing_tx_ids: HashSet<[u8; 32]> =
+                            wtxid_data.keys().cloned().collect();
+                        let aggregate_tx_ids: HashSet<[u8; 32]> = tx_ids.iter().cloned().collect();
+
+                        // If we have all the chunks, perform verification
+                        if aggregate_tx_ids.is_subset(&existing_tx_ids) {
+                            // Concatenate complete proof
+                            let complete_proof = tx_ids
+                                .iter()
+                                .filter_map(|k| wtxid_data.get(k).cloned())
+                                .fold(vec![], |mut p, c| {
+                                    p.extend(c);
+                                    p
+                                });
```

Comment:
Small nit but it looks nicer with flatten imo
```suggestion
                            let complete_proof =  tx_ids
            .iter()
            .filter_map(|k| wtxid_data.get(k).cloned())
            .flatten()
            .collect::<Vec<_>>();
```